packages/components/src/public/QueryInfo.ts

```diff
@@ -136,6 +138,14 @@ export class QueryInfo extends Record({
             columns = columns.set(rawColumn.fieldKey.toLowerCase(), QueryColumn.create(rawColumn));
         });
 
+        let altUpdateKeys;
```

Comment:
I would consider always initializing the `Set<string>()` so we don't have to deal with `undefined` for this value ever.